src/v/cluster/scheduling/allocation_strategy.cc

```diff
@@ -74,12 +75,16 @@ model::node_id find_best_fit(
     constraints,
   const std::vector<model::node_id>& possible_nodes,
   const allocation_state::underlying_t& nodes) {
-    uint32_t last_score = 0;
-    auto best_fit = possible_nodes.front();
+
+    vassert(!possible_nodes.empty(), "possible_nodes must not be empty");
```

Comment:
This condition was already ensured by the callers and the existing code relied on it, but let us encode it now as an assert.